src/java.base/share/classes/java/lang/reflect/Executable.java

```diff
@@ -778,4 +773,7 @@ public AnnotatedType[] getAnnotatedExceptionTypes() {
                 getGenericExceptionTypes(),
                 TypeAnnotation.TypeAnnotationTarget.THROWS);
     }
+
+    record ParameterData(@Stable Parameter[] parameters, boolean isReal) {}
```

Comment:
I recommend declaring this right next to `parameterData` field for the ease of maintenance, like how `EntrySet` in map implementations are declared right next to `entrySet` methods.